### PR TITLE
Refactor random seed

### PR DESCRIPTION
**Pull request by LucasAlegre**

### CODE REVIEW

File: morl_baselines/common/morl_algorithm.py [hunk 1]
```diff
@@ -160,19 +161,37 @@ def update(self) -> None:
...
+        self._np_random, _ = seeding.np_random(seed)
+
+    @property
+    def np_random(self) -> np.random.Generator:
+        """Returns the environment's internal :attr:`_np_random` that if not set will initialise with a random seed.
+
+        Returns:
+            Instances of `np.random.Generator`
+        """
+        if self._np_random is None:
```
Comment (ffelten) [inline morl_baselines/common/morl_algorithm.py:188]: it cannot be `None` from what I see in the `__init__` method

File: morl_baselines/common/morl_algorithm.py [hunk 2]
```diff
@@ -160,19 +161,37 @@ def update(self) -> None:
...
+            seed: (int): The seed to use for the random number generator
         """
         self.extract_env_info(env)
         self.device = th.device("cuda" if th.cuda.is_available() else "cpu") if device == "auto" else device
 
         self.global_step = 0
         self.num_episodes = 0
-        self.seed = None
+        self.seed = seed
+        self._np_random = None
```
Comment (ffelten) [inline morl_baselines/common/morl_algorithm.py:178]: remove this

File: benchmark/launch_experiment.py [hunk 3]
```diff
@@ -143,6 +144,8 @@ def main():
     args = parse_args()
     print(args)
 
+    seed_everything(args.seed)
```
Comment (ffelten) [inline benchmark/launch_experiment.py:147]: Can you document into seed_everything that it has to be called only once ?

File: morl_baselines/single_policy/ser/mo_q_learning.py [hunk 4]
```diff
@@ -72,8 +69,7 @@ def __init__(
         self.learning_rate = learning_rate
         self.id = id
         self.seed = seed
-        if self.seed is not None:
-            seed_everything(self.seed)
+        self.np_random, _ = seeding.np_random(self.seed)
```
Comment (ffelten) [inline morl_baselines/single_policy/ser/mo_q_learning.py:72]: Please write a small comment for why this case is different than other `MOAgent` (and in EUPG)

**Review by ffelten — CHANGES_REQUESTED** [review]
Small improvements and you can merge
